src/utils.rs

```diff
@@ -80,7 +80,41 @@ pub fn ident_at(root: &SyntaxNode, offset: usize) -> Option<CursorInfo> {
         }
     }?;
     let parent = ident.node().parent();
-    if let Some(attr) = parent.clone().and_then(Key::cast) {
+    if parent.clone().map(|x| x.kind() == rnix::SyntaxKind::NODE_INHERIT).unwrap_or(false) {
+        for n in parent.clone().unwrap().children_with_tokens() {
+            if n.kind() == rnix::SyntaxKind::NODE_INHERIT_FROM {
+                if let Some(node) = n.as_node() {
+                    for tok in node.children() {
+                        if tok.kind() == rnix::SyntaxKind::NODE_IDENT {
+                            return Some(CursorInfo {
+                                path: vec![tok.text().to_string()],
+                                ident: ident.clone(),
+                            })
+                        } else if tok.kind() == rnix::SyntaxKind::NODE_SELECT {
```

Comment:
Do we actually want to resolve the thing inside the parentheses?

```nix
{
    inherit (foo.bar) baz;
    #        ^^^^^^^
}
```

I mean we aren't actually importing it here, we're importing *from* it. So adding it here feels weird... I mean I get the intention, you wouldn't try to import items from somewhere that didn't exist, I guess. I don't know... maybe just a comment in the code like "hey this isn't actually something we're importing, but we're adding it because it makes sense" or something, could make this whole thing clearer?

Reply:
I'm not sure if I misunderstand your point: the `ident_at` function only returns the attribute-path or key to be completed. And we actually need to do that to know that `baz` is a key inside `foo.bar`.

Reply:
You know what, I'm an idiot, I completely misunderstood the point of your pull request. I didn't read properly and thought you wanted this:

```nix
let
  inherit (foo) bar baz;
in {
  a = ba
  #   ^^^ bar and baz would be completed.
}
```

but what you obviously wanted (I should have read more carefully!) was:

```nix
let
  foo = { bar = 1; baz = 2; };
in {
    inherit (foo) ba
    #             ^^^ bar and baz would be completed.
}
```

I hope that clears up why I added that confusing comment. Sorry :)